exec.js

```diff
@@ -44,15 +48,19 @@ var collectArgs = function(state) {
         // the argument (only for optional arguments)
         var activation = containers[i].getElementsByClassName("activation")[0];
         if ((activation && activation.checked) || !activation) {
-            var option = containers[i].getElementsByTagName("input");
-            // to handle dialog boxes
-            if (option.length == 1 && activation) {
-                var option = containers[i].getElementsByTagName("select")[0];
+            var option = containers[i].getElementsByClassName("arg-buttons")[0]
+                                        .getElementsByTagName("input");
+            // to handle drop down menu
+            if (!option.length) {
+                if (containers[i].getElementsByTagName("select")) {
+                    var option = containers[i].getElementsByTagName("select")[0];
+                }
+                console.log("something has gone wrong");
```

Comment:
When running the App, I see this on the console:

```
[7004:0922/233621:INFO:CONSOLE(58)] "something has gone wrong", source: /Users/sbasso/src/measurement-kit/network-meter/exec.js (58)
[7004:0922/233625:INFO:CONSOLE(58)] "something has gone wrong", source: /Users/sbasso/src/measurement-kit/network-meter/exec.js (58)
[7004:0922/233630:INFO:CONSOLE(58)] "something has gone wrong", source: /Users/sbasso/src/measurement-kit/network-meter/exec.js (58)
```

You probably want this to be printed only on `else`. Also, I'd rather throw an exception rather than a warning if `length` is zero and the control is not `select`, so that one notices there is an unexpected error and reports a bug to us.